runtime/altair/src/lib.rs

```diff
@@ -272,30 +133,20 @@ where
 	BalanceOfCollatorSelection<T>: From<u128>,
 	T::AccountId: From<sp_runtime::AccountId32>,
 	T::Keys: From<SessionKeys>,
-	[u8; 32]: From<<T as pallet_session::Config>::ValidatorId>,
+	<T as frame_system::Config>::AccountId: From<<T as pallet_session::Config>::ValidatorId>,
 {
 	fn on_runtime_upgrade() -> Weight {
 		let mut consumed: Weight = 0;
 
-		let current_validators_ids: Vec<[u8; 32]> = <pallet_session::Validators<T>>::get()
-			.into_iter()
-			.map(|who| Into::<[u8; 32]>::into(who))
-			.collect();
+		if VERSION.spec_version == IntegrateCollatorSelection::<T>::to_version() {
+			let current_validators = <pallet_session::Validators<T>>::get()
+				.into_iter()
+				.map(|who| who.into())
+				.collect();
 
-		let invulnerables = Self::into_T_tuple(current_validators_ids);
-		let candidates = Self::into_T_tuple(CANDIDATES.to_vec());
+			<pallet_collator_selection::Invulnerables<T>>::set(current_validators);
 
-		if VERSION.spec_version == IntegrateCollatorSelection::<T>::to_version() {
-			consumed +=
-				IntegrateCollatorSelection::<T>::inject_invulnerables(invulnerables.as_slice());
-			consumed +=
-				IntegrateCollatorSelection::<T>::inject_desired_candidates(DESIRED_CANDIDATES);
-			consumed += IntegrateCollatorSelection::<T>::inject_candidates(
-				candidates.as_slice(),
-				CANDIDACY_BOND.into(),
-			);
-			consumed +=
-				IntegrateCollatorSelection::<T>::inject_candidacy_bond(CANDIDACY_BOND.into());
```

Comment:
We still should inject `CandidacyBond` and `DesiredCandidates`. Or why do you want remove them? Not sure, If they are set correctly if we don`t go through genesis init.